pyracf/common/downstream_fatal_error.py

```diff
@@ -0,0 +1,66 @@
+"""Exception to use when IRRSMO00 returns with an error."""
+from typing import Union
+
+
+class DownstreamFatalError(Exception):
+    """
+    Raised IRRSMO00 returns with a return code of 8, indicating an error occured prior to request.
+    """
+
+    def __init__(
+        self,
+        xml_str: str,
```

Comment:
`xml_str` is a list of integers, no? In that case this should be called `return_reason_codes: List[int]` instead.

Reply:
good point. This was a relic of old structure.

Reply:
Now that I think about it, it may be even better to just extract these values from the list and make them individual arguments to the constructor:

```python
    def __init__(
        self,
        saf_return_code: int,
        racf_return_code: int,
        racf_reason_code: int,
        request_xml: bytes,
        run_as_userid: Union[str, None] = None,
        result_dictionary: dict = None,
    ) -> None:
```